### ISSUE TITLE
[DOC] installation instructions general and windows, updates/clarifications

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#1372]
This issue is about reproducing and potentially documenting installation of the developer version that seems to deviate from current installation instructions; and reproducing whether the current instructions, as they are written down, work properly (or not).

---
Via an environment mishap, I was forced to re-install my `sktime` development version on windows.

The vanilla workflow described in the installation instructions did not work - I had to do this (on the newest Windows version, with Anaconda):

first navigate to local `sktime` clone root folder, then:

```
conda create --name sktime python=3.8
conda activate sktime
conda install brotlipy
pip install -e .[all-extras]
pip install pytest
pip install esig
conda install -c conda-forge fbprophet
pip install numpy==1.19.5 --force-reinstall
```

Some takeaways, I think we need to update the install instructions:
* `fbprophet` causes problems, it needs to be singled out in a separate installation step. The instructions should be explicit, e.g., about installing it first separately, or later (which then makes the force-install `numpy` line necessary)
* setup.py extras do not include esig, which causes breakage in the dev version
* `pip install -e .[all-extras]` is actually what the developer user wants, but that line actually doesn't appear in the install instructions
* and what's that about `brotlipy`, not sure - I did this after following a random fix for an error message I got

**Comment by fkiraly** [comment]
Would also be good if someone could test setting up environment on windows under the current installation instructions.

I feel this also has to do with the binder fix of @corvusrabus which required a two-step approach, and possibly with work done by @freddyaboulton - perhaps you something wise to say about the above?

**Comment by freddyaboulton** [comment]
@fkiraly Regarding prophet, I don't think we need a two-step approach. I think we need to switch over to the `prophet` package as opposed to `fbprophet`. I think the problem is that `fbprophet` is quite old and they list `pystan>=2.14` as a requirement. This causes `pystan` 3.2.0 to be installed which is actually not compatible with prophet! Prophet only supports pystan 2. `prophet` has the correct pin of `pystan~=2.19.1.1`. 

I modified the setup.py to list prophet instead of prophet and then changed `fbprophet.py` to import `prophet` instead of `fbprophet` and I think everything is working ok. 

Can you give that a shot? I guess the alternative is to list `pystan` 2.19 as a requirement but I think just using `prophet` is cleaner.

Not sure what happened with `brotlipy`. I agree we should also list `esig` in `all_extras`. You can't even run the unit tests without it.

**Comment by fkiraly** [comment]
>I modified the setup.py

Where exactly? Don't see a new PR.

**Comment by freddyaboulton** [comment]
@fkiraly This is what I was talking about: https://github.com/alan-turing-institute/sktime/pull/1375

**Comment by fkiraly** [comment]
ah, thanks!

**Comment by TonyBagnall** [comment]
I'm going through this myself now, not sure what this warning is meant to mean 
![image](https://user-images.githubusercontent.com/9594042/141648282-ea0c3988-5845-44b4-8391-e9d7b9d2b501.png)

**Comment by TonyBagnall** [comment]
pip install -e .[all-extras] ends with

**Comment by TonyBagnall** [comment]
then this 
![image](https://user-images.githubusercontent.com/9594042/141648544-1ce935f9-ca82-445f-ae46-4d63f516aa9a.png)

**Comment by TonyBagnall** [comment]
going for option 2 above ...

**Comment by TonyBagnall** [comment]
I had to pip install cython before install -e

**Comment by TonyBagnall** [comment]
this is painful! Seems to try install numpy twice here? 
![image](https://user-images.githubusercontent.com/9594042/141658948-20359942-7057-48e2-9ffd-c71214535064.png)

**Comment by fkiraly** [comment]
@TonyBagnall, can you summarize the instrutions that worked for you and confirm your operating system?

Also, can you please clarify what you mean with option 1 or option 2?

**Comment by TonyBagnall** [comment]
see above comment by Aparna-Sakshi for option 1 and 2. I'll try sort this out on windows 10 over the next couple of weeks, install doesn't work well at all.

**Comment by TonyBagnall** [comment]
btw its not pip install -e .[all-extras] its pip install -e .[all_extras], you might want to edit the post above

**Comment by jasonlines** [comment]
Hello, I've just done a fresh install using @fkiraly's instructions above and it seems to be fine. i.e. 

```
conda create --name sktime python=3.8
conda activate sktime
conda install brotlipy
pip install -e .[all-extras]
pip install pytest
pip install esig
conda install -c conda-forge fbprophet
pip install numpy==1.19.5 --force-reinstall 
```

I'm on Windows and didn't need to do anything additional for `fbprophet` (it did complain during `pip install -e .[all-extras]` as expected, but I didn't need to do anything more than the `conda install -c conda-forge fbprophet` step to rectify that) so it doesn't look like I needed to follow the additional instructions from @Aparna-Sakshi and @TonyBagnall (but perhaps I just got lucky!)

**Comment by TonyBagnall** [comment]
hi jay, did you use all_extras? all-extras doesnt work.

**Comment by jasonlines** [comment]
ah yes sorry, I did use `all_extras`! 

Specifically, it was:
![image](https://user-images.githubusercontent.com/38794632/142216702-11da2ae1-7277-47ee-be31-4b32247abc9a.png)

**Comment by TonyBagnall** [comment]
and the cython works? I'm going to do it from scra

**Comment by jasonlines** [comment]
Yep, no problem with cython for me. It did fail on my first go because I hadn't installed a compiler yet, so I deleted the env, installed the community edition of Visual Studio with the C++ compiler, then tried again and it worked fine (verified by importing the dtw function into some code)

**Comment by jasonlines** [comment]
Update: I'm a dummy and forgot to downgrade NumPy at the end so cython _did_ work but Numba didn't. 

After downgrading NumPy on the last step it breaks the cythonised code because the binaries don't match for the version of NumPy that was used to compile the code vs. the downgraded version of NumPy that is then active and installed. 

The fix is to make sure NumPy is downgraded **before** installing cython/compiling the cython code. The steps I followed were:

```
conda create --name sktime python=3.8
conda activate sktime
conda install brotlipy
pip install -e .  
pip install numpy==1.19.5 --force-reinstall
pip install -e .[all_extras]
pip install pytest
pip install esig
conda install -c conda-forge fbprophet
```

The main changes to @fkiraly's original instructions are that `pip install -e .` is used initially without `all_extras` as this includes cython. You can then downgrade NumPy as the next instruction before installing with `all_extras` to install cython and compile any cython code - this doesn't affect NumPy as the 1.19.5 version still satisfies the minimum version requirements in setup. 

The instructions are the same otherwise and @TonyBagnall has run through the same successfully now

**Comment by TonyBagnall** [comment]
I didnt need conda install brotlipy

**Comment by fkiraly-shell** [comment]
from some company internal user testing:

* older VS build tools are no longer easily available, new version is 2022. Installation seems to work with 2022, i.e., MSVC v143, and does not need 2017 specifically anymore.
* installation guide should contain "validate" steps
* "should look like this" screenshots would be useful
* "make docs" is not available on Windows
* missing instruction to create a new environment (not using base or other)
* comment of python version should be repeated at start of developer set-up section
* install command should be `pip install --editable .[all-extras]` (as in my instructions at the top), this is still wrong! soft dependencies are not installed when following the standard workflow! needs to be fixed/added
* need to add clearer instructions in "need to install" first
* contradiction between instructions for contrib: no forking (dev install) vs forking (contributing guide)

**Comment by fkiraly** [comment]
I think this is addressed in #1672.
